### PR TITLE
Apply 'LWG3253 basic_syncbuf::basic_syncbuf() should not be explicit'

### LINKED ISSUE #100264: LWG3253: `basic_syncbuf::basic_syncbuf()` should not be explicit
**Link:** https://wg21.link/lwg3253

### PR DESCRIPTION
**Pull request by c8ef**
close: #100264

### CODE REVIEW

File: libcxx/docs/Status/Cxx20Issues.csv [hunk 1]
```diff
@@ -172,7 +172,7 @@
 "`3221 <https://wg21.link/LWG3221>`__","Result of ``year_month``\  arithmetic with ``months``\  is ambiguous","Belfast","|Complete|","8.0"
 "`3235 <https://wg21.link/LWG3235>`__","``parse``\  manipulator without abbreviation is not callable","Belfast","",""
 "`3246 <https://wg21.link/LWG3246>`__","What are the constraints on the template parameter of ``basic_format_arg``\ ?","Belfast","","","|format|"
-"`3253 <https://wg21.link/LWG3253>`__","``basic_syncbuf::basic_syncbuf()``\  should not be explicit","Belfast","",""
+"`3253 <https://wg21.link/LWG3253>`__","``basic_syncbuf::basic_syncbuf()``\  should not be explicit","Belfast","|Complete|","19.0"
```
Comment (Zingam) [inline libcxx/docs/Status/Cxx20Issues.csv:175]:
```suggestion
"`3253 <https://wg21.link/LWG3253>`__","``basic_syncbuf::basic_syncbuf()``\  should not be explicit","Belfast","|Complete|","20.0"
```

Comment (Zingam) [inline libcxx/docs/Status/Cxx20Issues.csv:175]:
```suggestion
"`3253 <https://wg21.link/LWG3253>`__","``basic_syncbuf::basic_syncbuf()``\  should not be explicit","Belfast","|Complete|","20.0"
```

File: libcxx/include/syncstream [hunk 2]
```diff
@@ -252,9 +254,9 @@ public:
   using streambuf_type = basic_streambuf<_CharT, _Traits>;
 
   // [syncstream.syncbuf.cons], construction and destruction
+  _LIBCPP_HIDE_FROM_ABI basic_syncbuf() : basic_syncbuf(nullptr) {}
```
Comment (Zingam) [inline libcxx/include/syncstream:257]: Do you have a test?

**Comment by Zingam** [comment]
Typically libc++ PRs are prefixed with `[libc++][syncstream] LWG....`.

**Comment by mordante** [comment]
I'm closing this since it's a duplicate of https://github.com/llvm/llvm-project/pull/99778, which has been approved.

**Pull request closed without merging**